content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1792 @@
+# Prepared Statement with Binding Methods
+
++++
+
+### Part 1: Sample Code Analysis
+
+:::
+
+/// type=REPL, readonly=true, filename=[Connection.php,Student.php]
+
+```php
+<?php
+    class Connection
+    {
+        const HOST = 'localhost';
+        const DB = 'LibraryDB';
+        const PORT = '5432';
+        const USERNAME = 'codestop';
+        const PASSWORD = 'Admin01';
+    
+        public static function getConnection()
+        {
+            try {
+                return new PDO("pgsql:host=" . self::HOST . ";port=". self::PORT . ";dbname=" . self::DB . ";user=" . self::USERNAME . ";password=" . self::PASSWORD);
+            } catch (Exception $e) {
+                throw new Exception("Unable to establish a connection."); 
+            }
+        }
+    }
+?>
+```
+
+```php
+<?php
+    require_once("./Connection.php");
+
+    $conn = Connection::getConnection();
+    $data = array(
+        'min_birth_date' => '1999-01-01',
+        'max_birth_date' => '2000-01-01'
+    );
+    
+    $sql = "SELECT * FROM students WHERE birth_date >= :min_birth_date AND birth_date < :max_birth_date";
+
+    $pstmt = $conn->prepare($sql);
+    $pstmt->bindParam(':min_birth_date', $data['min_birth_date'], PDO::PARAM_STR);
+    $pstmt->bindParam(':max_birth_date', $data['max_birth_date'], PDO::PARAM_STR);
+    $pstmt->execute();
+    $data = $pstmt->fetchAll();
+
+    foreach ($data as $row) {
+        echo $row['student_id'] . "\t";
+        echo $row['first_name'] . " ";
+        echo $row['last_name'] . "\t";
+        echo $row['birth_date'] . "\t";
+        echo $row['gender'] . "\n";
+    }
+?>
+```
+/// type=MS, answer=[1,2,3] 
+
+Execute the program. What are printed on lines 2, 3, and 4?
+
+- `1 John Smith 1999-02-10 M`
+
+- `6 Alisa Ells 1999-06-30 F`
+
+- `2 Samantha Danes 1999-10-12 F`
+
+- `Unable to establish a connection.`
+
+- `Successfully connected to the database.`
```

Comment:
I don't think the spaces will render in tabs ( `\t`). How about we use 4 spaces for each tab.